packages/client/components/AnalyticsPage.tsx

```diff
@@ -89,39 +89,39 @@ const TIME_TO_RENDER_TREE = 100
 
 const AnalyticsPage = () => {
   const atmosphere = useAtmosphere()
-  const [isGAInitialized, setIsGAInitialized] = useState(false)
+  if (gaMeasurementId) {
+    ReactGA.initialize(gaMeasurementId, {
+      gtagOptions: {
+        send_page_view: true
+      }
+    })
+  }
+
   useEffect(() => {
-    const initializeGA = async () => {
-      const res = await atmosphere.fetchQuery<AnalyticsPageQuery>(query)
-      if (!res) return
-      const {viewer} = res
-      const {id, segmentId} = viewer
-      ReactGA.initialize(gaMeasurementId, {
-        gtagOptions: {
-          send_page_view: true
-        }
-      })
-      ReactGA.set({
-        userId: id,
-        clientId: segmentId ?? getAnonymousId()
-      })
-    }
-    if (atmosphere.viewerId && !isGAInitialized && gaMeasurementId) {
-      initializeGA()
-        .catch()
-        .then(() => {
-          setIsGAInitialized(true)
+    const configGA = async () => {
+      if (!ReactGA.isInitialized || !isSegmentLoaded) {
+        return
+      }
+
+      const {viewerId} = atmosphere
+      if (viewerId) {
+        const res = await atmosphere.fetchQuery<AnalyticsPageQuery>(query)
+        if (!res) return
+        const {viewer} = res
+        const {id, segmentId} = viewer
+        ReactGA.set({
+          userId: id,
+          clientId: segmentId ?? getAnonymousId()
         })
+      } else {
+        ReactGA.set({
+          userId: null,
+          clientId: getAnonymousId()
```

Comment:
We need the `clientId` before login & after logout to attribute the entire user journey